src/ctap2/mod.rs

```diff
@@ -423,14 +424,32 @@ pub fn register<Dev: FidoDevice>(
         }
     }
 
+    // Client extension processing for credProtect:
+    // "When enforceCredentialProtectionPolicy is true, and credentialProtectionPolicy's value is
+    // [not "Optional"], the platform SHOULD NOT create the credential in a way that does not
+    // implement the requested protection policy. (For example, by creating it on an authenticator
+    // that does not support this extension.)"
+    if args.extensions.enforce_credential_protection_policy == Some(true)
+        && args.extensions.credential_protection_policy
+            != Some(CredentialProtectionPolicy::UserVerificationOptional)
+        && dev
+            .get_authenticator_info()
+            .map_or(false, |info| !info.supports_cred_protect())
```

Comment:
If the authenticator info isn't available, should we conservatively assume it doesn't support credential protection? (i.e. map to `true` rather than `false` here)

Reply:
yes! And a temporary `supports_cred_protect` bool would make this a lot more clear.